dspace-rest/src/main/java/org/dspace/rest/CollectionsResource.java

```diff
@@ -772,4 +872,59 @@ else if (!AuthorizeManager.authorizeActionBoolean(context, collection, action))
         }
         return collection;
     }
+
+	/**
+	 * Find item from DSpace database. It is encapsulation of method
+	 * org.dspace.content.Item.find with checking if item exist and if
+	 * user logged into context has permission to do passed action.
+	 *
+	 * @param context
+	 *            Context of actual logged user.
+	 * @param id
+	 *            Id of item in DSpace.
+	 * @param action
+	 *            Constant from org.dspace.core.Constants.
+	 *
+	 * @return DSpace Item found.
+	 *
+	 * @throws WebApplicationException
+	 *             Is thrown when item with passed id is not exists and if user
+	 *             has no permission to do passed action.
+	 */
+	private org.dspace.content.Item findItem(org.dspace.core.Context context, int id, int action)
+			throws WebApplicationException
+	{
+		org.dspace.content.Item item = null;
+		try
+		{
+			item = org.dspace.content.Item.find(context, id);
+
+			if (item == null)
+			{
+				context.abort();
+				log.warn("Item(id=" + id + ") was not found!");
+				throw new WebApplicationException(Response.Status.NOT_FOUND);
+			}
+			else if (!AuthorizeManager.authorizeActionBoolean(context, item, action))
+			{
+				context.abort();
+				if (context.getCurrentUser() != null)
+				{
+					log.error("User(" + context.getCurrentUser().getEmail() + ") has not permission to "
+							+ getActionString(action) + " item!");
```

Comment:
I see your point, I think I might have borrowed that directly from the existing 'findCollection' method and it ended up getting a bit distorted.